src/swaps/liqualityboost/lbspERC20toNative/LbspERC20toNative.js

```diff
@@ -0,0 +1,345 @@
+        quote.bridgeAsset
+      ].estimateFees({
+        network,
+        walletId,
+        asset,
+        txType: LbspERC20toNative.txTypes.SWAP,
+        quote: {
+          ...quote,
+          to: quote.bridgeAsset,
+          toAmount: quote.bridgeAssetAmount,
+          slippagePercentage
+        },
+        feePrices,
+        max
+      })
+      const totalFees = {}
+      for (const key in automatedMarketMakerFees) {
+        totalFees[key] = BN(automatedMarketMakerFees[key]).plus(liqualityFees[key])
+      }
+
+      return totalFees
+    }
+
+    return liqualityFees
+  }
+
+  async finalizeAutomatedMarketMakerAndStartLiqualitySwap({
+    swapLiqualityFormat,
+    swapAMMFormat,
+    network,
+    walletId
+  }) {
+    let result = await this.bridgeAssetToAutomatedMarketMaker[
+      swapAMMFormat.bridgeAsset
+    ].waitForSwapConfirmations({
+      swap: swapAMMFormat,
+      network,
+      walletId
+    })
+
+    if (result?.status === 'SUCCESS') {
+      result = await this.liqualitySwapProvider.newSwap({
+        network,
+        walletId,
+        quote: swapLiqualityFormat
+      })
+
+      return {
+        ...result,
+        ...swapLiqualityFormat,
+        toAmount: result.toAmount,
+        status: result.status
+      }
+    }
+  }
+
+  async performNextSwapAction(store, { network, walletId, swap }) {
+    let updates
+    const swapLiqualityFormat = {
+      ...swap,
+      from: swap.bridgeAsset,
+      fromAmount: swap.bridgeAssetAmount,
+      toAccountId: swap.fromAccountId,
+      slippagePercentage
+    }
+
+    const swapAutomatedMarketMakerFormat = {
+      ...swap,
+      to: swap.bridgeAsset,
+      toAmount: swap.bridgeAssetAmount,
+      slippagePercentage
+    }
+
+    if (swap.status === 'WAITING_FOR_SWAP_CONFIRMATIONS') {
+      updates = await withInterval(async () =>
+        this.finalizeAutomatedMarketMakerAndStartLiqualitySwap({
+          swapLiqualityFormat,
+          swapAMMFormat: swapAutomatedMarketMakerFormat,
+          network,
+          walletId
+        })
+      )
+    } else {
+      updates = await this.liqualitySwapProvider.performNextSwapAction(store, {
+        network,
+        walletId,
+        swap: swapLiqualityFormat
+      })
+    }
+
+    if (!updates && !LbspERC20toNative.lspEndStates.includes(swap.status)) {
+      updates = await this.bridgeAssetToAutomatedMarketMaker[
+        swap.bridgeAsset
+      ].performNextSwapAction(store, {
+        network,
+        walletId,
+        swap: swapAutomatedMarketMakerFormat
+      })
+    }
+
+    if (!updates) return
+
+    return {
+      ...updates,
+      // reset from and to assets and values
+      from: swap.from,
+      fromAmount: swap.fromAmount,
+      to: swap.to,
+      // keep `toAmount` (from updates object) only in case swap transitioned from AMM to LSP
+      toAmount: updates.status === 'INITIATED' ? updates.toAmount : swap.toAmount
```

Comment:
same as the above comment, regarding the returned properties as they are also use in timeline diagram if I am not mistaken

Reply:
This a specific case in which swap transitions from AMM to LSP. In this case due to slippage the `toAmount` seems to go a little bit lower. That's why the new value is kept.